pytorch_lightning/metrics/regression.py

```diff
@@ -187,3 +187,38 @@ def forward(self, pred: torch.Tensor, target: torch.Tensor) -> torch.Tensor:
             A Tensor with the rmsle loss.
         """
         return F.mse_loss(torch.log(pred + 1), torch.log(target + 1), self.reduction)
+
+
+class PSNR(Metric):
+    """
+    Computes the peak signal-to-noise ratio metric
+    """
+ 
+    def __init__(self, data_range: float = None, base: int = 10):
+        """
+        Args:
+            data_range: the range of the data. If None, it is determined from the data (max - min).
+            base: a base of a logarithm to use (default: 10)
+
+
+        Example:
+ 
+            >>> pred = torch.tensor([[0.0, 1.0], [2.0, 3.0]])
+            >>> target = torch.tensor([[3.0, 2.0], [1.0, 0.0]])
+            >>> metric = PSNR()
+            >>> metric(pred, target)
+            tensor([2.5527])
+        """
+        super().__init__(name='psnr')
+        self.data_range = data_range
+        self.base = torch.tensor([float(base)])
+
+    def forward(self, pred: torch.Tensor, target: torch.Tensor) -> torch.Tensor:
+        if self.data_range is None:
+            data_range = max(target.max() - target.min(), pred.max() - pred.min())
+        else:
+            data_range = torch.tensor(float(self.data_range))
+        mse = F.mse_loss(pred.view(-1), target.view(-1))
+        # numerical precision tricks
+        psnr_base_e = 2 * torch.log(data_range) - torch.log(mse)
+        return psnr_base_e * (10 / torch.log(self.base))  # change the logarithm basis
```

Comment:
You don't have to replicate the whole procedure again here. Just import from functional and use that one. :) Also add `reduction` parameter here.

Reply:
Refactored and added

BTW in case you didn't know, `pytest` says `'elementwise_mean'` is deprecated, it might be smart to refactor the default to `'mean'`. I left it that way to match rohitgr7's PR.

Reply:
Nice! Also, can you fix the pep8 issues mentioned above by @pep8speaks.

Reply:
@rohitgr7 I'm confused about the missing whitespace. I thought that was a bad thing :|

Reply:
hmm, don't know about that :| I never use an extra `,` after the last element in the list.

Reply:
Nvm I figured what was wrong